### ISSUE TITLE
cuDF hash_values returns incorrect values 

### ISSUE THREAD

**Issue by dorisnvidia** [rapidsai/cudf#3857]
**Describe the bug**
cudf series hash_values() returns different hash values for the same string in different cudf columns. 

**Steps/Code to reproduce bug**
```
import cudf
gdf = cudf.DataFrame()
gdf = cudf.DataFrame()
gdf['src_str'] = ['ESP_PSR_100FG@@ESP_PSR_10PLANT', 'ESP_PSR_100FG@@ESP_PSR_10PLANT'];
gdf['dst_str'] = ['ESP_PSR_100FG@@ESP_PSR_10DC',    'ESP_PSR_100FG@@ESP_PSR_10PLANT']
gdf['src_hash'] = gdf['src_str'].hash_values()
gdf['dst_hash'] = gdf['dst_str'].hash_values()
gdf
```

**Expected behavior**
In Row 0, src and dst are the different string, but hash_values() generated the same value. and in Row 1, src and dst are the same string, but different hash values are generated. 

**Environment overview (please complete the following information)**
docker
docker pull rapidsai/rapidsai:cuda10.1-runtime-centos7

**Environment details**
run on Prom cluster, pull the docker container. 

**Additional context**
we found a few issues with cuDF hash functions when working with a client on cugraph data processing. First we tried:
gdf['src_hash'] = gdf['src_str'].str.hash()
but the resulted unique count of gdf['src_hash'] is slightly different from that of the original string column, which is wrong. I could upload a dataset for you to reproduce the error, if you want to, as I cannot create an easier reproducer. But please just focus on the hash_values error for now.

**Comment by jlowe** [comment]
I suspect this is the same issue as #1473.  It looks like it's hashing the string category index value rather than the string contents.

**Comment by kkraus14** [comment]
Was about to report the same, this is related to #1473 but at the Python layer we need to update the bindings to use the new libcudf APIs so will leave this in place.

**Comment by dorisnvidia** [comment]
thanks for the response! do we have a ET for the bug fix?

**Comment by kkraus14** [comment]
> thanks for the response! do we have a ET for the bug fix?

Likely mid-0.13 so ~2-3 weeks would be my guess.

**Comment by dorisnvidia** [comment]
great, thank you!

**Comment by kkraus14** [comment]
Resolved as of 0.13.
